src/yaml/pull_parser.cr

```diff
@@ -122,6 +126,75 @@ class YAML::PullParser
     anchor
   end
 
+  def read_null
+    expect_scalar_style LibYAML::ScalarStyle::PLAIN
+    unless YAML::NULL_VALUES.includes?(self.value)
+      raise "Expected null, not '#{self.value}'", *location
+    end
+    read_next
+    nil
+  end
+
+  def read_bool
+    expect_scalar_style LibYAML::ScalarStyle::PLAIN
+    value = if YAML::TRUE_VALUES.includes?(self.value)
+              true
+            elsif YAML::FALSE_VALUES.includes?(self.value)
+              false
+            else
+              raise "Expected boolean, not '#{self.value}'", *location
+            end
+    read_next
+    value
+  end
+
+  def read_int
+    expect_scalar_style LibYAML::ScalarStyle::PLAIN
+    if value = self.value.to_s.gsub('_', "").to_i64?(prefix: true)
+      value
```

Comment:
This single variable makes no sense. The entire `if` statement could be replaced with this:
```crystal
value = self.value.to_s.gsub('_', "").to_i64?(prefix: true)
raise "Expected integer not '#{self.value}'", *location unless value
````